server/server.go

```diff
@@ -381,39 +381,10 @@ func New(ctx context.Context, config *Config) (*Server, error) {
 	s.restore()
 	s.cleanupSandboxesOnShutdown(ctx)
 
-	hostIP := net.ParseIP(config.HostIP)
-	if hostIP == nil {
-		// First, attempt to find a primary IP from /proc/net/route deterministically
-		hostIP, err = knet.ChooseBindAddress(nil)
-		if err != nil {
-			// if that fails, check if we can find a primary IP address unambiguously
-			allAddrs, err2 := net.InterfaceAddrs()
-			if err2 != nil {
-				return nil, errors.Wrapf(err, "Failed to read InterfaceAddrs after failing to choose bind address")
-			}
-
-			// we are hoping for exactly 1 possible IP
-			numPossibleIPs := 0
-			// adapted from: https://stackoverflow.com/a/31551220
-			for _, addr := range allAddrs {
-				if ipnet, ok := addr.(*net.IPNet); ok && !ipnet.IP.IsLoopback() {
-					if ipnet.IP.To4() != nil {
-						numPossibleIPs++
-						hostIP = ipnet.IP
-						if numPossibleIPs > 1 {
-							break
-						}
-					}
-				}
-			}
-
-			// If there is no clear primary IP, we should fail
-			if numPossibleIPs != 1 {
-				return nil, errors.Wrapf(err, "Failed to find one IP address after failing to choose bind address")
-			}
```

Comment:
I don't see this check for the numPossibleIPs != 1 anymore, is that no longer a concern or do I need another cup of tea?

Reply:
yeah the main concern there was crio choosing a different ip than the kubelet (so we wanted crio to choose on with no ambiguity). turns out that's not really necessary, and the kubelet chooses the first that works